src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblyConsoleLogger.cs

```diff
@@ -2,12 +2,50 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Concurrent;
 using System.Text;
 using Microsoft.Extensions.Logging;
 using Microsoft.JSInterop;
 
 namespace Microsoft.AspNetCore.Components.WebAssembly.Services
 {
+    /// <summary>
+    /// A provider of <see cref="WebAssemblyConsoleLogger{T}"/> instances.
+    /// </summary>
+    public class WebAssemblyConsoleLoggerProvider : ILoggerProvider
+    {
+        private ConcurrentDictionary<string, WebAssemblyConsoleLogger<object>> _loggers;
+        private IJSInProcessRuntime _jsRuntime;
+        private bool _disposed;
+
+        /// <summary>
+        /// Creates an instance of <see cref="WebAssemblyConsoleLoggerProvider"/>.
+        /// </summary>
+        /// <param name="options">The options to create <see cref="WebAssemblyConsoleLogger"/> instances with.</param>
+        public WebAssemblyConsoleLoggerProvider(IJSInProcessRuntime jsRuntime)
+        {
+            _loggers = new ConcurrentDictionary<string, WebAssemblyConsoleLogger<object>>();
+            _jsRuntime = jsRuntime;
+        }
+
+        /// <inheritdoc />
+        public ILogger CreateLogger(string name)
+        {
+            return _loggers.GetOrAdd(name, loggerName => new WebAssemblyConsoleLogger<object>(name, _jsRuntime));
+        }
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            if (!_disposed)
+            {
+                _loggers = null;
```

Comment:
Hmm, this is a strange. We generally don't null out fields after use. Is there a good reason to do this?

Reply:
It's true that we don't, but there's also no harm in it tbh, specially since this is a singleton

Reply:
If we are nulling them out, could do so unconditionally without tracking `_disposed`

Reply:
> but there's also no harm in it tbh,

You can no longer make fields `readonly` which results in a poorer pattern. Removing it seems fine.

Reply:
Ok yes, agreed.